spec/invoker.e2e.spec.js

```diff
@@ -0,0 +1,131 @@
+const {TextEncoder} = require('util');
+const startInvoker = require('../lib/invoker');
+const outputSignalCustomEqual = require('./helpers/output-signal-custom-equality');
+const newClient = require('./helpers/grpc-client');
+const {
+    newInputFrame,
+    newInputSignal,
+    newOutputFrame,
+    newOutputSignal,
+    newStartFrame,
+    newStartSignal
+} = require('./helpers/factories');
+
+describe('invoker =>', () => {
+    const textEncoder = new TextEncoder();
+    let server;
+    let address;
+    let client;
+
+    afterEach((done) => {
+        server.tryShutdown(done);
+        client.close();
+    });
+
+    [
+        {interactionType: 'streaming', functionUri: '../samples/streaming-square/streaming-square.js'},
+        {interactionType: 'request-reply', functionUri: '../samples/request-reply-square/square.js'},
+    ].forEach((testCase) => {
+
+        describe(`with a ${testCase.interactionType} square function =>`, () => {
+
+            beforeEach(() => {
+                jasmine.addCustomEqualityTester(outputSignalCustomEqual);
+                ({server, address} = tryStartInvoker(testCase.functionUri));
+                client = newClient(address);
+            });
+
+            it('successfully invokes the function', (done) => {
+                const inputs = [
+                    newStartSignal(newStartFrame(['application/json'])),
+                    newInputSignal(newInputFrame(0, 'text/plain', textEncoder.encode('2'))),
```

Comment:
```suggestion
                    newInputSignal(newInputFrame(0, 'application/json', textEncoder.encode('2'))),
```

That you can square a string in JavaScript is a hack we shouldn't encourage.